### PR TITLE
stable-2.1 | more backports

### PR DESCRIPTION
**Pull request by snir911**
Second round of backports, since #2226 (not merged yet)
( I didn't backport https://github.com/kata-containers/kata-containers/pull/2310 as backporting the vendored part seems to be too intrusive, if it make sense to have it without this patch i can add it, otherwise it may be better to [?not]backport it in separate)

b559177f docs: Remove kata-proxy and binaries reference
b233f0af static-checks: Restrict static checks to go 1.15 and 1.16
26dacbcb runtime: Update vendored code (according to branch)
aaab71fb runtime: Hot-plug virtio-mem device on PCI bridge
3f341a3f static-checks: Check the vendored code
45420163 build: Add `make vendor`
7447875d agent-ctl: Add `make vendor`
4ed576f6 trace-forwarder: Add `make vendor`
d9271cf2 agent: Add `make vendor`
a2314976 runtime: Add `make vendor`
f684098a runtime: Add `make handle_vendor`
cd42e063 qemu/arm: remove nvdimm/"ReadOnly" option on arm64
12d27aa4 snap: do not export agent version
a957c6e4 virtiofsd: fix the issue of missing stop virtiofsd
a33918e2 kata-deploy: Use the correct image for kata-deploy
71bd0f5d config: Fix description for OCI hooks
0aa857de shimv2: fix the issue of kata-runtime exec failed
a645bd05 snap: fixed snap aarch64 qemu patches dir in snapcraft.yaml file
ad19d6ec snap: Remove QEMU before clone
0ce41eb7 docs: fixed kata-deploy path for kata logs with fluentd doc
56d44082 mount: fix the issue of missing check file exists

### CODE REVIEW

**Comment by fidencio** [comment]
Looking at the static check failure makes me wonder if we have 1474f085c7023e1eb77210132dcf24a972880560 backported on the tests side.

**Review by fidencio — APPROVED** [review]
thanks for working on those backports, @snir911!

**Review by devimc — APPROVED** [review]
thanks @snir911

**Comment by devimc** [comment]
osbuilder is failing 

```
08:48:24 Creating '_chrony' system user/group for the chronyd daemon…
08:48:24 chfn: PAM: System error
08:48:24 adduser: `/usr/bin/chfn -f Chrony daemon _chrony' returned error code 1. Exiting.
08:48:24 dpkg: error processing package chrony (--configure):
08:48:24  installed chrony package post-installation script subprocess returned error exit status 1
```

@Jakob-Naucke  should @snir911 backport https://github.com/kata-containers/kata-containers/pull/2150 ?

**Comment by Jakob-Naucke** [comment]
> osbuilder is failing
> 
> ```
> 08:48:24 Creating '_chrony' system user/group for the chronyd daemon…
> 08:48:24 chfn: PAM: System error
> 08:48:24 adduser: `/usr/bin/chfn -f Chrony daemon _chrony' returned error code 1. Exiting.
> 08:48:24 dpkg: error processing package chrony (--configure):
> 08:48:24  installed chrony package post-installation script subprocess returned error exit status 1
> ```
> 
> @Jakob-Naucke should @snir911 backport #2150 ?

@devimc I've definitely seen this, and that might fix it. The change would be less intrusive with merely backporting the updated installation method of `chrony`. IIRC, at least with an Ubuntu 20.04 guest, it only worked with Podman on Fedora-like hosts.

**Comment by snir911** [comment]
@Jakob-Naucke i agree it's too intrusive, i added https://github.com/kata-containers/kata-containers/pull/2351/commits/04b4bd1e1311561ee4e85823c534a899b8cd4a22 does it make sense?
If it will fail i suppose I'll have to backport also https://github.com/kata-containers/tests/pull/3636/commits/26cef9d550dbee1ced48acbcaafaf594cc653b18

File: tools/osbuilder/rootfs-builder/ubuntu/rootfs_lib.sh [hunk 1]
```diff
@@ -59,27 +59,18 @@ build_rootfs() {
...
-	fi
 
 	${PKG_MANAGER} --variant=minbase \
 		--arch=${ARCHITECTURE}\
 		--include="$PACKAGES" \
 		${OS_NAME} \
 		${ROOTFS_DIR}
 
-	chroot $ROOTFS_DIR ln -s /lib/systemd/systemd /usr/lib/systemd/systemd
+	[ -n "${EXTRA_PKGS}" ] && chroot $ROOTFS_DIR apt-get install -y ${EXTRA_PKGS}
```
Comment (Jakob-Naucke) [inline tools/osbuilder/rootfs-builder/ubuntu/rootfs_lib.sh:73]: @snir911 almost LGTM -- technically, this line does not substitute the other line like the diff implies, but the added line belongs to the new installation method of chrony (which you want to backport) and the removed line belongs to the upgrade to 20.04 (which you didn't want to backport).

...I'm not happy with my phrasing. I think it should just be:

```sh
chroot $ROOTFS_DIR ln -s /lib/systemd/systemd /usr/lib/systemd/systemd
[ -n "${EXTRA_PKGS}" ] && chroot $ROOTFS_DIR apt-get install -y ${EXTRA_PKGS}
```

**Review by fidencio — APPROVED** [review]
lgtm, assuming happy tests!

**Comment by snir911** [comment]
> + chroot /rootfs ln -s /lib/systemd/systemd /usr/lib/systemd/systemd
> + '[' -n ' chrony chrony' ']'
> + chroot /rootfs apt-get install -y chrony chrony
Build timed out (after 5 minutes). Marking the build as aborted.

@Jakob-Naucke familiar? 

> ci/jenkins_job_build.sh: line 165: .ci/ci_job_flags.sh: No such file or directory

@wainersm this oddly happens in stable-2.1, and none of your related patches was backported, i think somehow [adding the depends-on](https://github.com/kata-containers/kata-containers/pull/2351/commits/a0ce77f1c5e9b39a92d2788e55d0ea7446e0a181) is causing this, although in the logs it seems to pull the correct stable branch PR and not main (jenkins-vfio-host:ubuntu18-04-vm:fedora32)

**Comment by Jakob-Naucke** [comment]
> ```chroot /rootfs ln -s /lib/systemd/systemd /usr/lib/systemd/systemd
> '[' -n ' chrony chrony' ']'
> chroot /rootfs apt-get install -y chrony chrony
> Build timed out (after 5 minutes). Marking the build as aborted.
> ```
> 
> @Jakob-Naucke familiar?

@snir911 hmmm. If this rings a bell at all, it would be very quiet and from quite some time ago. It's interesting that this doesn't always happen. I'm tempted to suggest to just retest.

**Comment by fidencio** [comment]
We appreciate the contributions to the stable-2.1 branch, but as Kata Containers 2.2.0 has been released we're moving the effort to maintain **only** the stable-2.2 for now.

Thanks a lot for the help, and we're closing this PR!

**Pull request closed without merging**